### ISSUE TITLE
Swagger Docs | Filter

### ISSUE THREAD

**Issue by flogruber** [fschick/FilterExpressionCreator#10]
Hello,

I cam across a bug when I wanted to implement filtering in my api.  

**Programm.cs**
```c#
builder.Services.AddControllers()
    .AddFilterSupport();

builder.Services.AddSwaggerGen(opt =>
{
    opt.AddFilterSupport();
});
```

**Controller.cs**
```c#
[HttpGet("", Name = nameof(GetEquipments))]
[ProducesResponseType(typeof(ListEquipmentDtoResponse), StatusCodes.Status200OK)]
public async Task<IActionResult> GetEquipments(Guid tenantId, [FromQuery] EntityFilter<Equipment> filter)
{
   ...
}

[HttpGet("{id:Guid}", Name = nameof(GetEquipment))]
[ProducesResponseType(typeof(EquipmentDtoResponse), StatusCodes.Status200OK)]
[ProducesResponseType(typeof(EquipmentDtoResponse), StatusCodes.Status404NotFound)]
public async Task<IActionResult> GetEquipment(Guid id)
{
   ...
}
```

My intention is to implement this filter only for endpoints that also require a filter and not, for example, where an explicit resource is requested by id.  
If I now start the api with this implementation, I get the following error message when generating the swagger docs. 

```
Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware[1]
      An unhandled exception has occurred while executing the request.
      Swashbuckle.AspNetCore.SwaggerGen.SwaggerGeneratorException: Failed to generate Operation for action - demo.api.Controllers.EquipmentController.GetEquipment (demo.api). See inner exception
       ---> System.NullReferenceException: Object reference not set to an instance of an object.
         at Plainquire.Filter.Swashbuckle.OpenApiParameterExtensions.IsEntityFilterParameter(ApiParameterDescription description)
         at Plainquire.Filter.Swashbuckle.Filters.EntityFilterParameterReplacer.<>c.<GetEntityFilterReplacements>b__3_3(ValueTuple`2 openApi)
         at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
         at System.Linq.Lookup`2.Create(IEnumerable`1 source, Func`2 keySelector, IEqualityComparer`1 comparer)
         at System.Linq.GroupedEnumerable`2.GetEnumerator()
         at System.Linq.Enumerable.SelectEnumerableIterator`2.ToList()
         at Plainquire.Filter.Swashbuckle.Filters.EntityFilterParameterReplacer.GetEntityFilterReplacements(OpenApiOperation operation, OperationFilterContext context)
         at Plainquire.Filter.Swashbuckle.Filters.EntityFilterParameterReplacer.Apply(OpenApiOperation operation, OperationFilterContext context)
         at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.<GenerateOperationAsync>b__21_0(OpenApiOperation operation, OperationFilterContext filterContext)
         at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GenerateOperationAsync(ApiDescription apiDescription, SchemaRepository schemaRepository, Func`3 parametersGenerator, Func`3 bodyGenerator, Func`3 applyFilters)
         --- End of inner exception stack trace ---
         at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GenerateOperationAsync(ApiDescription apiDescription, SchemaRepository schemaRepository, Func`3 parametersGenerator, Func`3 bodyGenerator, Func`3 applyFilters)
         at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GenerateOperationAsync(ApiDescription apiDescription, SchemaRepository schemaRepository)
         at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GenerateOperationsAsync(IEnumerable`1 apiDescriptions, SchemaRepository schemaRepository)
         at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GeneratePathsAsync(IEnumerable`1 apiDescriptions, SchemaRepository schemaRepository, Func`3 operationsGenerator)
         at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GeneratePathsAsync(IEnumerable`1 apiDescriptions, SchemaRepository schemaRepository)
         at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GetSwaggerAsync(String documentName, String host, String basePath)
         at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
         at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
         at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(HttpContext context)
```

Have I by any change forgotten something for the implementation that the docs can be generated?
Or does filtering have to be implemented for all endpoints?

Thank you!

**Comment by fschick** [comment]
I assume it is a bug. I try to reproduce it this evening.

**Comment by flogruber** [comment]
I have found the root cause.  
I have a parameter `tenantId`in the route that is not used in the one function.  

```C#
[Route("api/c/tenants/{tenantId:guid}/[controller]")]
```

The initial endpoint looked like this:
```C#
[HttpGet("{id:Guid}", Name = nameof(GetEquipment))]
[ProducesResponseType(typeof(EquipmentDtoResponse), StatusCodes.Status200OK)]
[ProducesResponseType(typeof(EquipmentDtoResponse), StatusCodes.Status404NotFound)]
public async Task<IActionResult> GetEquipment(Guid id)
{
   ...
}
``` 

I changed it to 
```C#
[HttpGet("{id:Guid}", Name = nameof(GetEquipment))]
[ProducesResponseType(typeof(EquipmentDtoResponse), StatusCodes.Status200OK)]
[ProducesResponseType(typeof(EquipmentDtoResponse), StatusCodes.Status404NotFound)]
public async Task<IActionResult> GetEquipment(Guid tenantId, Guid id)
{
   ...
}
```

After this change, the swagger docs are generated.  

However, I still have a question about this.  
The query parameters are implemented during an import with NSwag without default value/null.

Is it possible to define the query parameters in the swagger documentation with default values?  
Instead of settings the following parameter when importing with NSwag?  
```
/GenerateOptionalParameters:true
```

**Comment by fschick** [comment]
Exception fixed with [Release 5.1.1](https://github.com/plainquire/plainquire/releases/tag/5.1.1).

**Comment by fschick** [comment]
Regarding your question to define default values for parameters in the swagger documentation: If you don't like to modify your controllers, it's possible to modify the generated swagger.json using filters. See [Schema Filters](https://github.com/domaindrivendev/Swashbuckle.AspNetCore?tab=readme-ov-file#apply-schema-filters-to-specific-types) and [Operation Filters](https://github.com/domaindrivendev/Swashbuckle.AspNetCore?tab=readme-ov-file#extend-generator-with-operation-schema--document-filters). For more complex sample see `Entity*ParameterReplacer.cs` files in https://github.com/plainquire/plainquire.
